diff -ruN lib/binfhe-base-scheme.cpp lib.new/binfhe-base-scheme.cpp
--- lib/binfhe-base-scheme.cpp	2024-01-03 13:16:42.193299327 +0100
+++ lib.new/binfhe-base-scheme.cpp	2024-01-03 13:42:39.946267425 +0100
@@ -63,6 +63,7 @@
     skNPoly.SetFormat(Format::EVALUATION);
 
     ek.BSkey = ACCscheme->KeyGenAcc(RGSWParams, skNPoly, LWEsk);
+    ek.skN = skNPoly;
 
     return ek;
 }
diff -ruN lib/binfhecontext.cpp lib.new/binfhecontext.cpp
--- lib/binfhecontext.cpp	2024-01-03 13:16:42.193299327 +0100
+++ lib.new/binfhecontext.cpp	2024-01-03 13:43:54.227496602 +0100
@@ -142,6 +142,11 @@
         { STD192Q_4,         { 34,     4096,         980, 2048, 1 << 17, STD_DEV,     16,    1 << 12,  32,    10,  UNIFORM_TERNARY} },
         { STD256Q_4,         { 27,     4096,        1625, 4096, 1 << 21, STD_DEV,     16,    1 <<  6,  32,    10,  UNIFORM_TERNARY} },
         { SIGNED_MOD_TEST,   { 28,     2048,         512, 1024,   PRIME, STD_DEV,     25,    1 <<  7,  23,    10,  UNIFORM_TERNARY} },
+        { SET_I, {28, 2048, 600, 1024, 1 << 17, STD_DEV, 32, 1 << 7, 23, 10, UNIFORM_TERNARY}},
+        { SET_II, {28, 2048, 600, 1024, 1 << 17, STD_DEV, 32, 1 << 7,23, 10, UNIFORM_TERNARY}},
+        { SET_III, {36, 2048, 448, 1024, 1 << 17, STD_DEV, 32, 1 << 12,23, 10, UNIFORM_TERNARY}},
+        { SET_IV, {42, 4096, 600, 2048, 1 << 24, STD_DEV, 128, 1 << 14,23, 20, UNIFORM_TERNARY}},
+        { SET_V, {52, 4096, 700, 2048, 1 << 24, STD_DEV, 32, 1 << 26,23, 20, UNIFORM_TERNARY}}
     });
     // clang-format on
 
